var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -109,6 +106,16 @@ class Libfabric(AutotoolsPackage):
              sha256='3b78d0ca1b223ff21b7f5b3627e67e358e3c18b700f86b017e2233fee7e88c2e',
              placement='fabtests', when='@1.4.2')
 
+    conflicts('@1.9.0', when='platform=darwin',
+              msg='This distribution is missing critical files')
+
+    def url_for_version(self, version):
+        return '{base}/v{version}/libfabric-{version}.tar.{ext}'.format(
+            base='https://github.com/ofiwg/libfabric/releases/download',
+            version=str(version),
+            ext=('bz2' if version >= Version('1.8') else 'gz')
+        )
```

Comment:
Alternatively, we could switch the checksums for all previous releases to download the tar.bz2 version and get rid of this function. That would make the package simpler.